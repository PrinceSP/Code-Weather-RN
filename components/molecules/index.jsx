import Header from './Header'
import CustomUnits from './CustomUnits'
import Checkbox from './CheckBox'
import HourlyWeather from './HourlyWeather'
import LocationPlace from './LocationPlace'
import SearchHeader from './SearchHeader'

export {Header,CustomUnits,Checkbox,HourlyWeather,LocationPlace,SearchHeader}
