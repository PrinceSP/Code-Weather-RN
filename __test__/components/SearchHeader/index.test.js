import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import {SearchHeader} from '../../../components';

describe('SearchHeader',()=>{
  const updateFunction = jest.fn();
  const mockTitle = "Search"

  it('renders correctly', () => {
    const componentRender = render(
      <SearchHeader updateModal={updateFunction} updateQuery={updateFunction} />
    );
    expect(componentRender).toMatchSnapshot();
  });

  it('should update query correctly when typing in TextInput', () => {
    const { getByPlaceholderText } = render(
      <SearchHeader updateModal={() => {}} updateQuery={updateFunction} />
    );
    const searchInput = getByPlaceholderText('Search');
    fireEvent.changeText(searchInput, 'test query');
    expect(updateFunction).toHaveBeenCalledWith('test query');
  });
})
