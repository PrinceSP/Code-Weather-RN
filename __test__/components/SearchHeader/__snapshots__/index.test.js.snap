// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#47484955",
        "borderRadius": 20,
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 6,
        "width": "90%",
      }
    }
  >
    <Text />
    <TextInput
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#aaa"
      style={
        {
          "color": "#fff",
          "fontSize": 9,
          "marginLeft": 10,
          "width": "90%",
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
        }
      }
    >
      X
    </Text>
  </View>
</View>
`;
